### PR TITLE
Support stream cancellation on TS client

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/SignalR/issues/898

### CODE REVIEW

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Observable.ts [hunk 1]
```diff
@@ -10,16 +10,39 @@ export interface Observer<T> {
...
+    observer: Observer<T>;
+
+    constructor(subject: Subject<T>, observer: Observer<T>) {
+        this.subject = subject;
+        this.observer = observer;
+    }
+
+    public dispose(): void {
+        let index: number = this.subject.observers.indexOf(this.observer);
+        if (index > -1) {
```
Comment (mikaelm12) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Observable.ts]: Would an equality check be faster?

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Observable.ts [hunk 2]
```diff
@@ -10,16 +10,39 @@ export interface Observer<T> {
     complete?: () => void;
 }
 
+export class ObserverDisposable<T> {
```
Comment (mikaelm12) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Observable.ts]: Should this just be called Subscription?

Comment (mikaelm12) [reply client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Observable.ts]: I think it makes more sense as Subscription. It's the result of calling subscribe. It's the Rx convention as well I believe

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by mikaelm12** [comment]
Looks pretty good to me but I'll let someone more comfortable with typescript do the final sign off 😄

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts [hunk 3]
```diff
@@ -310,4 +317,11 @@ export class HubConnection {
             arguments: args,
         };
     }
+
+    private createCancelInvocation(id: string): CancelInvocation {
+        return <CancelInvocation>{
```
Comment (anurse) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts]: Do you need the cast? I thought TypeScript would be smart enough to infer this. If it's necessary though, keep it.

**Pull request merged**